### PR TITLE
Listing all lots on home page, instead of all building permits

### PR DESCRIPTION
**Pull request by fabiomcosta**

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -34,104 +34,29 @@ import {
...
-                              {/*19374 BANFIELD PL */}
-                            </Td>
-                            <Td>{item.parcel_number}</Td>
-                            <Td>{item.permit_address}</Td>
-                            <Td>{item.contractor_phone_number}</Td>
-                            <Td isNumeric>{item.zip}</Td>
+                          <Tr key={item.number}>
+                            <Td>{item.number}</Td>
+                            <Td>{item.address}</Td>
+                            <Td>?</Td>
```
Comment (fabiomcosta) [inline pages/index.js:55]: only showing lot number, address and hopefully in the future we'll be able to figure out how to give a "status" of the lot like "complete" or pre-drywall etc...
I'm not sure how to do that though.. maybe Bruno will know

File: pages/index.js [hunk 2]
```diff
@@ -182,7 +107,107 @@ export default function Home() {
                   </Flex> */}
                 </TableContainer>
               </CardBody>
-            </Card>}
+            </Card>
+}
+
+function useCityData() {
```
Comment (fabiomcosta) [inline pages/index.js:113]: centralized the data fetching and massaging on this hook

File: pages/index.js (hunk 1 above)
Comment (agnelnieves) [reply pages/index.js:55]: Sounds good! Added it here: https://github.com/agnelnieves/summerdale-permits/issues/6

File: pages/index.js [hunk 3]
```diff
@@ -182,7 +107,107 @@ export default function Home() {
...
+  const lotData = useMemo(() => {
+    const lots = data.reduce((acc, permit) => {
+      const lotNumber = Number(permit.parcel_number.slice(-5, -1))
+      if (acc.hasOwnProperty(lotNumber)){
+        acc[lotNumber].permits.push(permit)
+      } else {
+        acc[lotNumber] = {
+          number: lotNumber,
+          address: permit.permit_address,
+          // we don't actually need this
```
Comment (agnelnieves) [inline pages/index.js:146]: The permit?

File: pages/index.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 // --- React/Nextjs
-import { useEffect, useState, useCallback } from 'react'
+import { useEffect, useState, useCallback, useMemo } from 'react'
```
Comment (agnelnieves) [inline pages/index.js:2]: 🔥

**Review by agnelnieves — APPROVED** [review]
Looks good!

File: pages/index.js (hunk 3 above)
Comment (fabiomcosta) [reply pages/index.js:146]: we don't actually use the `permits` array that I'm creating here on the homepage, but created it anyway 🤷

**Pull request merged**